### DISCUSSION TITLE
Multi user on one port

### DISCUSSION THREAD

**Discussion by hosseinnoob** [FranzKafkaYu/x-ui#234] [Ideas]
pls add limits for each user in one port like https://github.com/hossinasaadi/x-ui
thanks

**Comment by ashkan-moghadasi** [comment]
Yes it's nice feature . There is maximum 10 user constraint on each port .I think this is cause trouble when you are limited by a few ports by cdn

**Comment by fer000n** [comment]
hi dude
how we can use multi user in same port? is any tutorial for this?
thanks bro

**Comment by Jabi-james** [comment]
to do that he has to recompile the xray-core and add this future on it  every time there is an update available..
but it is a very useful tool

**Comment by armansoheili** [comment]
I think its gonna be perfect to have no limit user per same port, 10 user is not enough

**Reply by FranzKafkaYu** [reply]
> I think its gonna be perfect to have no limit user per same port, 10 user is not enough

In China someone would use this tool for selling proxies to others,and this may cause problem to me( Government pressure).so this restriction can prevent this tool being exploit. And there are other panels can create unlimited users.you can try these panels.
